packages/frontend/src/components/MetricQueryData/UnderlyingDataModal.tsx

```diff
@@ -22,6 +29,15 @@ const UnderlyingDataModal: FC<Props> = () => {
                 minWidth: '500px',
             }}
         >
+            {isUnderlyingDataModalOpen &&
+                track({
```

Comment:
We should only track the events in the button onClick so we can capture the page & section context and also avoid duplicate events